### ISSUE TITLE
How to integrate on_message decorated functions from resume-chat and bigquery?

### ISSUE THREAD

**Issue by scheung38** [Chainlit/chainlit#748]
resume-chat:

```
@cl.on_message
async def on_message(message: cl.Message):
   
    await chain(message.content)
     
    memory = cl.user_session.get("memory")  # type: ConversationBufferMemory

    runnable = cl.user_session.get("runnable")  # type: Runnable

    res = cl.Message(content="")
  
    async for chunk in runnable.astream(
        {"question": message.content},
        config=RunnableConfig(callbacks=[cl.LangchainCallbackHandler()]),
    ):
        await res.stream_token(chunk)
       
    await res.send()
    
    memory.chat_memory.add_user_message(message.content)
    memory.chat_memory.add_ai_message(res.content)
```
    
bigquery:

```
@cl.on_message
async def main(message: cl.Message):
    await chain(message.content)    
```  

So that there is history in the context of a text-SQL

```
@cl.on_message
async def on_message(message: cl.Message):
   
    await chain(message.content) <--- added here
     
    memory = cl.user_session.get("memory")  # type: ConversationBufferMemory

    runnable = cl.user_session.get("runnable")  # type: Runnable

    res = cl.Message(content="")
    
    async for chunk in runnable.astream(
        {"question": message.content},
        config=RunnableConfig(callbacks=[cl.LangchainCallbackHandler()]),
    ):
        await res.stream_token(chunk)
       
    await res.send()
    
    memory.chat_memory.add_user_message(message.content)

    memory.chat_memory.add_ai_message(res.content)
```

But two issues:
    
    1. First time query:
    
    `get top 5 rows of invoicenumber, amount, payment date, payment status from payment report between Nov 2023 to Jan 2024
    `
    
![Screenshot 2024-02-17 at 14 24 04](https://github.com/Chainlit/chainlit/assets/1273014/eefe5b0a-f4c7-418a-9a78-0f3b827d3c05)

    
    Would provide output from text-SQL but since empty [] first response would be text [] with message: 
    
    I'm sorry, but I am not able to access specific data like payment reports from a database
    
    Even though it had correctly generated text-to-SQL intermediate result
    
    
    2. Follow up query:
    out of this, which has highest payment amount, does not have context as expected:
    
![Screenshot 2024-02-17 at 14 25 49](https://github.com/Chainlit/chainlit/assets/1273014/3fef3ad5-7a31-4723-8054-589a398f52a7)

So basically how to get persistence from resume-chat and bigQuery text-to-SQL to work with history in this on_message thanks
